test/forward_backward_compatibility/check_forward_backward_compatibility.py

```diff
@@ -249,44 +255,105 @@ def process_version_map(version_map):
     return output
 
 
+def is_core_aten_op(schema) -> bool:
+    # Check if the schema is a core ATen op
+    if "::" not in schema.name:
+        return False
+    _, _, tags = torch._C._get_operation_overload(schema.name, schema.overload_name)
+    return Tag.core in tags
+
+
 def check_bc(existing_schemas):
     new_schema_dict = load_schemas_to_dict()
     version_map = process_version_map(torch._C._get_operator_version_map())
     is_bc = True
     broken_ops = []
     for existing_schema in existing_schemas:
         if allow_listed(existing_schema):
-            print("schema: ", str(existing_schema), " found on allowlist, skipping")
+            logging.info("schema: %s found on allowlist, skipping", existing_schema)
             continue
         if has_valid_upgraders(existing_schema, version_map):
-            print("schema: ", str(existing_schema), " has valid upgrader, skipping")
+            logging.info("schema: %s has valid upgrader, skipping", existing_schema)
             continue
-        print("processing existing schema: ", str(existing_schema))
+        logging.info("processing existing schema: %s", existing_schema)
         matching_new_schemas = new_schema_dict.get(existing_schema.name, [])
         found = False
         for matching_new_schema in matching_new_schemas:
             if matching_new_schema.is_backward_compatible_with(existing_schema):
                 found = True
                 break
         if not found:
-            print(
+            logging.warning(
                 "Can NOT find backward compatible schemas after changes "
-                "for schema {} from the following candidates:\n[\n{}\n]".format(
-                    str(existing_schema),
-                    "\n\t".join(str(s) for s in matching_new_schemas),
-                )
+                "for schema %s from the following candidates:\n[\n%s\n]",
+                str(existing_schema),
+                "\n\t".join(str(s) for s in matching_new_schemas),
             )
             # TODO Print out more details about why candidates don't match.
             broken_ops.append(str(existing_schema))
             is_bc = False
     if is_bc:
-        print("Found backward compatible schemas for all existing schemas")
+        logging.info("Found backward compatible schemas for all existing schemas")
     else:
-        print(
+        logging.warning(
             "The PR is introducing backward incompatible changes to the "
             "operator library. Please contact PyTorch team to confirm "
             "whether this change is wanted or not. \n\nBroken ops: "
-            "[\n\t{}\n]".format("\n\t".join(broken_ops))
+            "[\n\t%s\n]",
+            "\n\t".join(broken_ops),
+        )
+    return is_bc
+
+
+def check_core_aten_ops_bc(existing_schemas):
```

Comment:
What's the difference between this and the one above? They seem to be very similar?

Reply:
We don't look at the allowlist is the biggest difference. This will fail on any BC breaking changes on core ATen op. Of course they can still merge by overriding the CI job failure but I guess that's the best we can do?

Reply:
Why isn't the check for core API not part of the above function when we're about to use the allowlist? Would that reduce code duplication?